docs/ADR-5-cardano-testnet-node-configuration-file.md

```diff
@@ -0,0 +1,39 @@
+# Status
+
+📜 Proposed 2024-09-10
+
+# Context
+
+There is no canonical way to create a node configuration file. `cardano-cli` offers this possibility as part of the bigger [create-cardano](https://github.com/IntersectMBO/cardano-cli/blob/551d9b9f2f244e0d681bf03eaa6d985565ac3a5b/cardano-cli/test/cardano-cli-golden/files/golden/help/latest_genesis_create-cardano.cli#L49) command, but this is ad-hoc.
+
+Programmatic users proceed as follows instead:
+
+* Scripts like [mkfiles.sh](https://github.com/IntersectMBO/cardano-node/blob/51a034a51c5cefdd6ab4b9ff1e71710cf0c96643/scripts/babbage/mkfiles.sh#L85) and [jobs.nix](https://github.com/input-output-hk/cardano-parts/blob/0abf510e0ed70fda4e6ad3ae71632c20d09f135a/flakeModules/jobs.nix#L253) copy an existing template.
+* [cardano-testnet](https://github.com/IntersectMBO/cardano-node/blob/51a034a51c5cefdd6ab4b9ff1e71710cf0c96643/cardano-testnet/src/Testnet/Defaults.hs#L340) uses a hardcoded Haskell value.
+
+# Decision
+
+## Add a new command in the CLI
+
+Introduce a new `conway genesis check-node-config file` command in `cardano-cli`. It will have the following options:
+
+```
```

Comment:
Why list the options again? It should just check the existence of filepaths and that the genesis file hashes are correct. Have a look at the other fields that it may make sense to have checks for as well but to re-enter the file paths doesn't make sense imo. Parameterizing it on the era makes sense because things can change era to era and notifying the user what they may be missing in an upcoming era would be useful.

Reply:
Yes, actually this was an earlier design and the actual WIP work ([this branch](https://github.com/IntersectMBO/cardano-cli/tree/smelc/check-node-config)) has the following API:

```
cardano-cli debug check-node-configuration
  --node-configuration-file FILEPATH
  [--fix-configuration-file]
```

Updating the ADR accordingly right now :+1: